packages/x-data-grid/src/components/GridPagination.tsx

```diff
@@ -17,6 +17,7 @@ import {
 
 const GridPaginationRoot = styled(TablePagination)(({ theme }) => ({
   maxHeight: 'calc(100% + 1px)', // border width
+  flexGrow: 1,
```

Comment:
I've changed the approach here to avoid 1px shrinking for virtual scroller area: https://app.argos-ci.com/mui/mui-x/builds/25055/113076748